lib/chef/knife/bootstrap/windows-chef-client-msi.erb

```diff
@@ -107,12 +107,15 @@ goto install
 :install
 @rem Install Chef using chef-client MSI installer
 
-<% url="https://www.chef.io/chef/download?p=windows&pv=%MACHINE_OS%&m=%MACHINE_ARCH%" -%>
-<% url += latest_current_windows_chef_version_query -%>
-@set "REMOTE_SOURCE_MSI_URL=<%= url %>"
+<% default_url = "https://www.chef.io/chef/download?p=windows&pv=%MACHINE_OS%&m=%MACHINE_ARCH%" -%>
+<% default_url += latest_current_windows_chef_version_query -%>
+<% cs_url = msi_url.empty? ? default_url : msi_url -%>
+<% ps_url = msi_url.empty? ? default_url + "&DownloadContext=PowerShell" : msi_url -%>
+
+@set "REMOTE_SOURCE_MSI_URL=<%= cs_url %>"
+@set "REMOTE_SOURCE_MSI_PS_URL=<%= ps_url %>"
```

Comment:
there are a few places where we put %REMOTE_SOURCE_MSI_URL% into log output when something fails. I don't know why we pass "&DownloadContext=PowerShell" but maybe it would be important if that made it to the log output too.

For example, someone could say a URL is failing for them, and we try it and we can't reproduce it, but that's because REMOTE_SOURCE_MSI_PS_URL is being used but REMOTE_SOURCE_MSI_URL is being printed.